src/crepe/system/InputSystem.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include "../facade/SDLContext.h"
+#include "../types.h"
+#include "../util/OptionalRef.h"
+
+#include "System.h"
+
+namespace crepe {
+
+class Camera;
+class Button;
+class Transform;
+
+/**
+ * \brief Handles the processing of input events like mouse and keyboard interactions.
+ *
+ * This system processes events such as mouse clicks, mouse movement, and keyboard
+ * actions. It is responsible for detecting interactions with UI buttons and
+ * passing the corresponding events to the registered listeners.
+ */
+class InputSystem : public System {
+public:
+	using System::System;
+
+	/**
+     * \brief Updates the system, processing all input events.
+     * This method processes all events and triggers corresponding actions.
+     */
+	void update() override;
+
+private:
+	//! Stores the last position of the mouse when the button was pressed.
+	std::pair<int, int> last_mouse_down_position{INFINITY, INFINITY};
+
+	//! Stores the last mouse button pressed.
+	MouseButton last_mouse_button = MouseButton::NONE;
+	//
+	//! The tolerance in game units for detecting a mouse click.
+	const int click_tolerance = 5;
```

Comment:
Toleratie voor wat?

Reply:
hoe ver de muis mag bewegen om de klik te registreren zodat als je de muis inhoud en ergens anders naartoe sleept en dat los laat het geen klik meer is

Reply:
Nu staat het wel duidelijker uitgelegd in de doxygen comment maar het is inderdaad nog niet duidelijk of het in game units of pixels is. Ik vind zelf beide een slechte oplossing omdat je het bij game units afhankelijk van de camera zoom maakt en bij pixels afhankelijk van scherm DPI maakt. Biedt SDL iets van DPI-onafhankelijke schermcoördinaten? Anders mogen het van mij pixels blijven omdat niemand van ons een HiDPI monitor gebruikt.

Reply:
niels had het er over dat sdl inderdaad een manier heeft om dit automatisch te doen, maar ik weet niet precies of deze nu gebruikt worden. de laatste keer dat ik met niels sprak hoefde ik hiervoor zelf niks meer te doen

Reply:
Maak er een float van en dan documenteren we later dat deze (en andere) waarden in HUD-ruimte zijn